### ISSUE TITLE
Do not pass down internal props

### ISSUE THREAD

**Issue by smirea** [ctrlplusb/react-jobs#7]
Currently `jobInitState` and `onJobProcessed` are passed down to its children. 
This is not ideal if your children pass down the props:

```jsx
import {withJob} from 'react-jobs/ssr'

@withJob(() => Promise.resolve())
class Foo extends PureComponent {
    render () {
        return <div {...this.props}>React will complain endlessly</div>;
    }
}
```

**Comment by ctrlplusb** [comment]
Well spotted!

**Comment by ctrlplusb** [comment]
@smirea - released in v0.4.0, thanks!
